### PR TITLE
Automatically strip leading/trailing whitespace in input strings.

### PR DESCRIPTION
**Pull request by jgadling**
### Summary:
- **What:** Enable pydantic automatic leading/trailing whitespace stripping feature

### Notes:
Pydantic docs are here: https://pydantic-docs.helpmanual.io/usage/model_config/#options
We've already encountered some issues with samples with leading/trailing tabs and spaces in their identifier names, and keeping extra whitespace is never the right behavior for this app.

### Checklist:
- [x] I merged latest `<base branch>`
- [x] I manually verified the change
- [x] I added labels to my PR
- [ ] I tested in multiple browsers
- [x] I added relevant unit tests
- [ ] I have notified others of changes they need to make locally (migrations, jobs, package updates, etc)

### CODE REVIEW

File: src/backend/aspen/api/schemas/base.py [hunk 1]
```diff
@@ -4,7 +4,12 @@
 
 
 class BaseRequest(BaseModel):
-    pass
+    class Config:
+        """Extra configuration options"""
+
+        anystr_strip_whitespace = (
+            True  # remove leading/trailing whitespace from strings
```
Comment (vincent-czi) [inline src/backend/aspen/api/schemas/base.py:11]: Why is this in parens? Is it just an auto-formatter thing? It wanted to stay under 80chars or something? My first thought was there was some kind of tuple thing going on for the config, but I realize that's not the case now.

**Review by vincent-czi — APPROVED** [review]
Seems reasonable! Looks good, thanks for the test!

File: src/backend/aspen/api/schemas/base.py (hunk 1 above)
Comment (lvreynoso) [reply src/backend/aspen/api/schemas/base.py:11]: In this case the parens are just decorative and show an expression is written over multiple lines. Parentheses in Python can indicate a tuple if there's a comma separating a sequence, but in a single item tuple you would need a comma (e.g.`my_tuple = (item1,)`). They can also indicate a generator expression if it uses the comprehension syntax (`x for x in y if some_cond(x)`).

**Review by lvreynoso — APPROVED** [review]

File: src/backend/aspen/api/schemas/base.py (hunk 1 above)
Comment (jgadling) [reply src/backend/aspen/api/schemas/base.py:11]: the formatting tool didn't like my comment making the line super long so it wrapped the value onto another line with parens.

**Pull request merged**